src/Kovaleva.HomeWork-4/Task.cs

```diff
@@ -5,24 +5,49 @@ namespace Kovaleva.HomeWork_4
     public class Task
     {
         private readonly string Id;
-
+        private DateTime startDate;
+        private DateTime endDate;
         private string GetId()
         { return Guid.NewGuid().ToString().ToUpper().Substring(0, 5); }
 
+        private DateTime ConvertDateFromString(string value)
+        {
+            DateTime result;
+            try
```

Comment:
it is always better to avoid exceptions whenever possible (because exception handling is a quite resource intensive operation), so TryParse is more reasonable here

Reply:
У нас просто тема занятия была в т.ч. Исключения, поэтому это здесь просто как пример использования Try ..catch..

Reply:
Понял, в общем да, сама обработка исключений правильная, но пример не очень удачный, т.к. является плохой практикой.

Сравните например время выполнения с разными подходами 
 public static void Main()
        {            
            var sw = new Stopwatch();
            var dates = Enumerable.Repeat("invalid", 10000).Concat(new [] {"2021-01-01"}).ToArray();
            sw.Start();
            foreach (var d in dates)
            {
                //if (DateTime.TryParse(d, out DateTime date))
                if (TryParse(d, out DateTime date))
                Console.WriteLine(date);
            }
            sw.Stop();
            Console.WriteLine(sw.ElapsedMilliseconds);
        }

        private static bool TryParse(string value, out DateTime date)
        {
            date = DateTime.Now;
            try
            {
                date = DateTime.Parse(value);
            }
            catch (Exception)
            {
                return false;
            }
            return true;
        }